src/pack.ts

```diff
@@ -159,11 +159,8 @@ export async function pack(this: EsbuildServerlessPlugin) {
 
   // package each function
   await Promise.all(
-    buildResults.map(async ({ func, functionAlias, bundlePath }) => {
-      const name = `${this.serverless.service.getServiceName()}-${
-        this.serverless.service.provider.stage
-      }-${functionAlias}`;
-
+    buildResults.map(async ({ func, bundlePath }) => {
+      const fnName = func.name;
```

Comment:
@samchungy 
Please consider this PR, in `v1.10.7` I made the opposite change due to some issue with custom lambda naming, maybe you could workaround this so both cases would work?

Reply:
Nice good thing I requested your review 😊. I think it'll just be adjusting `copyExtra` or this one to use the same syntax.

Reply:
Okay @floydspace  so I looked at the above linked issue and it looks like they were using https://www.npmjs.com/package/serverless-pseudo-parameters which is now deprecated as Serverless Framework 1.5.0 does this behaviour natively. It looks like Serverless also parses this before we package which is nice which means we shouldn't have the same issues as before.

However, of course if we continue with this serverless-pseudo-parameters will not work, but in my opinion that sounds okay to me as it is deprecated. I tested this with `${aws:region}`, `${aws:accountId}` as well as a `${self:custom.name}` and it appears to work fine.

![image](https://user-images.githubusercontent.com/18017094/147030628-1cef99cc-a1c6-478d-8519-5ee1f8fc9a72.png)

Option 2:

If we would like to keep that behaviour I could always change this line

https://github.com/floydspace/serverless-esbuild/blob/fb8345d3f596fad4eb0452780e1cb4572cc09b9f/src/index.ts#L375-L377

to be eg.

```typescript
 const destFileName = path.resolve( 
   path.join(this.buildDirPath, `__only_${`${this.serverless.service.getServiceName()}-${ 
         this.options.stage || serverless.service.provider.stage 
       }-${functionAlias}`}`, filename) 
 ); 
 ```

Reply:
Okay @floydspace I've replaced it with the `functionAlias` which is how the Serverless Framework packages zips so we should have no compatibility issues unless someone is doing something which the Serverless Framework does not support 😆.

I've also refactored a few confusing `fnName` variables to `functionAlias` around the codebase as well as changed `for ... in this.functions` with `for ... of Object.entries(this.functions)`